io/src/main/java/smile/io/CSV.java

```diff
@@ -147,10 +147,13 @@ private DataFrame read(Reader reader, int limit) throws IOException {
             for (CSVRecord record : csv) {
                 Object[] row = new Object[fields.length];
                 for (int i = 0; i < fields.length; i++) {
-                    String s = record.get(i).trim();
-                    if (!s.isEmpty()) {
-                        row[i] = parser.get(i).apply(s);
-                    }
+					String recordI = record.get(i);
+					if (recordI != null) {
```

Comment:
Why is it possible null?

Reply:
CSV convert missing value to null.

Reply:
This is not necessary. We handle missing value correctly. You got issue here because you specify wrong data type. For double with missing values, it should be `DoubleObject`. BTW, you don't need specify the schema in most type as we automatically infer csv schema.